chef/automate
# Sending chef-server actions as events to the event-feed-service

### :nut_and_bolt: Description: What code changed, and why?
Migrating all the chef-server actions from the config-mgmt-service's "actions-*" indexes to the event-feed-service. This places all the events in one service. This allows other services to easily add events to the event feed. This simplifies the event feed service to not split requests between two services. 

### :bug: Bug fixed
Project update and project tagging on ingestion for Chef actions were using the incorrect field. It was using the "remote_hostname" but it should have been using the "service_hostname". These are both defined here https://github.com/chef-boneyard/chef-rfc/blob/master/rfc077-mode-agnostic-data-collection.md. 
This PR fixes the problem and re-ingests the action which fixes any incorrect project tags. 

Another bug was found that prevented the event-feed-service's events from filtering correctly on event types. components/event-feed-service/pkg/feed/feed.go

### :chains: Related Resources

#444 

### :athletic_shoe: How to Build and Test the Change
1. `start_all_services`
1. Add some actions with `chef_load_actions 25`
1. To ensure the actions are in the old indexes run `es_list_indices`. Then ensure there are indexes that start with "actions-".
1. Open the event feed UI page take a snapshot or remember how it looks.  
1. Update to the PR changes with: `rebuild components/event-feed-service && rebuild components/config-mgmt-service && rebuild components/ingest-service && rebuild components/automate-gateway && rebuild components/automate-deployment/`
1. To ensure the migration worked correctly, open the event feed page, and ensure the events added above have all the correct information.
1. Run `es_list_indices` again to ensure all the "action-" indexes are removed. 

### :white_check_mark: Checklist
